src/lib/components/Instances/GenerateInstances.svelte

```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
+	import { clickOutside } from '$lib/clickOutside';
+	import Button from '$lib/components/ui/Button.svelte';
+	import { Sparkle, Sparkles } from 'lucide-svelte';
+	import { fade } from 'svelte/transition';
+
+	let { similar, createInstances } = $props<{
+		similar: boolean;
+		createInstances: (instruction: string, count: number, similar: boolean) => void;
+	}>();
+
+	let showGenerateOptions = $state(false);
+	let generationInstruction = $state('');
+	let generationCount = $state(5);
+</script>
+
+<div class="relative" use:clickOutside={() => (showGenerateOptions = false)}>
+	<Button
+		onclick={() => (showGenerateOptions = !showGenerateOptions)}
+		title={similar ? 'Generate Similar' : 'Generate'}
+		classNames="text-yellow-400"
+	>
+		{#if similar}
+			<Sparkles class="h-5 w-5 transition" />
+		{:else}
+			<Sparkle class="h-5 w-5 transition" />
+		{/if}
+	</Button>
+	{#if showGenerateOptions}
+		<div
+			class="absolute right-0 top-10 z-20 w-max rounded border bg-white p-3"
+			onclick={(e) => e.stopPropagation()}
+			onkeydown={(e) => {
+				if (e.key === 'Escape') {
+					showGenerateOptions = false;
+				}
+			}}
+			role="button"
+			tabindex="0"
+			transition:fade={{ duration: 100 }}
+		>
+			<input
+				placeholder="Describe the new data."
+				bind:value={generationInstruction}
+				onkeydown={(e) => {
+					if (e.key === 'Enter') {
+						showGenerateOptions = false;
+						createInstances(generationInstruction, generationCount, similar);
+					}
+				}}
+				class="mb-2 w-full"
+				type="text"
+			/>
+			<div class="flex items-center gap-2">
+				<input bind:value={generationCount} type="range" min="1" max="10" />
+				<Button
+					classNames="shrink-0 min-w-52 justify-center"
+					onclick={() => {
+						showGenerateOptions = false;
+						createInstances(generationInstruction, generationCount, similar);
+					}}
+				>
+					Generate {generationCount} instances
```

Comment:
![2024-02-21 at 15 34 49](https://github.com/end-of-text/endoftext/assets/5690524/0edbc2be-3aeb-4f46-9f5f-c46aaf769fa4)

should probably say "generate 1 instance"

Reply:
ah sorry this reverted somehow.